lib/App/MoarVM/Debug.rakumod

```diff
@@ -342,11 +355,26 @@ sub caller(Int() $handle --> Nil) {
     say $result.&to-json(:pretty);
 }
 
-sub clearbp(Str() $file, Int() $line --> Nil) {
+multi sub clearbp(Int() $id --> Nil) {
+    say "Breakpoint with this ID ($id) does not exist" and return unless defined %breakpoints{$id};
+    my ($file, $line) = %breakpoints{$id}<file line>;
+    clearbp $file, $line, $id;
+}
+
+multi sub clearbp(Str() $file, Int() $line, Int() $id? is copy --> Nil) {
+    if !defined $id {
+        for %breakpoints.kv -> $k, $v {
+            if $v { if $v<file> eq $file and $v<line> eq $line { $id = $k; last; } }
+        }
+        say "No breakpoint like that ($file:$line) exists" and return if not defined $id
+    }
+    %breakpoints{$id} = Nil;
+
     my $result := remote
       "clearing breakpoint for $file:$line",
       { $remote.clear-breakpoints($file, $line) }
-    say $result.&to-json(:pretty);
+    
+    say "Deleted breakpoint for $file:$line with ID $id";
```

Comment:
i'm worried that this will misbehave when the user adds more than one breakpoint on the same file:line; moar will clear all breakpoints for a given file:line when you send the clear breakpoints request, but the above code only removes the first entry it sees in the %breakpoints hash

Reply:
(identifying breakpoints only by a filename+line number is a weakness of the protocol design, IMO)